client/src/components/RequestsView.js

```diff
@@ -48,6 +49,22 @@ export default class RequestsView extends Component {
             confirmedTimes: []
         }
         this.handleStatusUpdate = this.handleStatusUpdate.bind(this)
+        this.handleOffsetChange = this.handleOffsetChange.bind(this)
+    }
+
+    async handleOffsetChange(offset) {
+        if (this.state.timeOffset !== offset) {
+            window.location.reload()
```

Comment:
weirdly I'm ok with refreshing here - timezone shouldn't need to be changed that frequently, so a onetime refresh shouldn't be problematic

The only reason I don't like refresh in the requestModal is that it pulls you out of the modal